backends/intel_gpu/kernels/elementwise_kernel.cc

```diff
@@ -0,0 +1,230 @@
+                          phi::DenseTensor* out) {
+  show_kernel("ElementWise-ONEDNN type=" << dnn_support::type2String<T>::name());
+
+  auto* q = static_cast<sycl::queue*>(const_cast<void*>(dev_ctx.stream()));
+
+  if (!q) {
+
+  }
+
+  using namespace dnnl;
+  using tag = memory::format_tag;
+  using dt = memory::data_type;
+
+  auto eng = dnnl::sycl_interop::make_engine(q->get_device(), q->get_context());
+  auto engine_stream = dnnl::sycl_interop::make_stream(eng, *q);
+
+  dnnl::memory::dims dims_x = x.dims();
+  dnnl::memory::dims dims_y = y.dims();
+  dnnl::memory::dims dims_out = out->dims();
+
+  phi::update_broadcast(dims_x,dims_y,axis);
+
+ auto md_x = memory::desc(dims_x, dnn_support::toDnnType<T>::type, dnn_support::dims2Tag(dims_x));
+
+ auto md_y = memory::desc(
+     dims_y, dnn_support::toDnnType<T>::type, dnn_support::dims2Tag(dims_y));
+ auto md_out = memory::desc(dims_out,
+                            dnn_support::toDnnType<T>::type,
+                            dnn_support::dims2Tag(dims_out));
+
+ auto x_mem = memory(md_x, eng, x.data<T>());
+ auto y_mem = memory(md_y, eng, y.data<T>());
+
+ auto out_data = dev_ctx.template Alloc<T>(out);
+
+ auto out_mem = memory(md_out, eng, out_data);
+
+ auto oper_desc = binary::desc(algorithm::binary_mul, md_x, md_y, md_out);
+ auto prim_desc = binary::primitive_desc(oper_desc, eng);
+ auto prim = binary(prim_desc);
+
+ std::unordered_map<int, memory> binary_args;
+ binary_args.insert({DNNL_ARG_SRC_0, x_mem});
+ binary_args.insert({DNNL_ARG_SRC_1, y_mem});
+ binary_args.insert({DNNL_ARG_DST, out_mem});
+
+ prim.execute(engine_stream, binary_args);
+ engine_stream.wait();
+
+}
+
+template <typename T>
+void MultiplyOneDNNKernel(const phi::Context& dev_ctx,
+                            const phi::DenseTensor& x,
+                            const phi::DenseTensor& y,
+                            phi::DenseTensor* out) {
+  int axis = -1;
+  MultiplyOneDNNRawKernel<T>(dev_ctx, x, y, axis, out);
+}
+
+
+
+template <typename T>
+void MultiplyMainRaw(const phi::Context& dev_ctx,
+                     const phi::DenseTensor& x,
+                     const phi::DenseTensor& y,
+                     int axis,
+                     phi::DenseTensor* out) {
+
+   if constexpr (std::is_same<T, float>::value
+                || std::is_same<T, int32_t>::value
+              //  || std::is_same<T,double>::value
+                  ) {
+     MultiplyOneDNNRawKernel<T>(dev_ctx, x, y, axis, out);
+   } else {
+     MultiplyRawKernelGPU<T>(dev_ctx, x, y, axis, out);
+   }
+}
+  template <typename T>
+  void MultiplyMain(const phi::Context& dev_ctx,
+                    const phi::DenseTensor& x,
+                    const phi::DenseTensor& y,
+                    phi::DenseTensor* out) {
+
+      int axis = -1;
+      MultiplyMainRaw<T>(dev_ctx, x, y, axis, out);
+
+
+  }
+
+
+
+}  // namespace custom_kernel
+
+
+
+
+
+
+
+             PD_BUILD_PHI_KERNEL(multiply_raw,
```

Comment:
code style not correct, please run  "pip install pre-commit && pre-commit install" under PaddleCustomDevice, and then run "git commit -m xxxx" to format all codes styles.